### PR TITLE
Export multiple path in dataproxy lib and minor fixes

### PR DESCRIPTION
**Pull request by zatteo**

### CODE REVIEW

**Review by Ldoppea — APPROVED** [review]
Suggestion: you can also refactor the main `index.js` with something like
```
export * from './ui'
export * from './api'
```
This would prevent future mistakes so developers won't be tempted to add new imports to it and forget to add them in the `/api` or `/ui` indexes.

File: packages/cozy-dataproxy-lib/src/api.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { SearchEngine } from './search/SearchEngine'
+export { DataProxyProvider, useDataProxy } from './dataproxy/DataProxyProvider'
```
Comment (paultranvan) [inline packages/cozy-dataproxy-lib/src/api.ts:2]: Are `DataProxyProvider` and `useDataProxy` necessary here? I think they are currently only used inside the ui search

Comment (zatteo) [reply packages/cozy-dataproxy-lib/src/api.ts:2]: I can move them if they are more linked to the ui part.

Comment (paultranvan) [reply packages/cozy-dataproxy-lib/src/api.ts:2]: I'm not sure, as I guess the provider can be seen as an api, more than a ui concept. Let's keep it like this, but I think you can remove the `useDataProxy` export

Comment (zatteo) [reply packages/cozy-dataproxy-lib/src/api.ts:2]: Agree, even it will be used by ui components, it's more an api concept 👍

**Review by paultranvan — APPROVED** [review]
Thanks a lot!

**Pull request merged**